src/core/cgroup.c

```diff
@@ -2769,6 +2781,36 @@ static void unit_add_to_cgroup_oom_queue(Unit *u) {
                 log_error_errno(r, "Failed to enable cgroup oom event source: %m");
 }
 
+static int unit_check_cgroup_events(Unit *u) {
+        int r, i;
+        char *values[2] = {};
+
+        assert(u);
+
+        r = cg_get_keyed_attribute(SYSTEMD_CGROUP_CONTROLLER, u->cgroup_path, "cgroup.events", STRV_MAKE("populated", "frozen"), values);
+        /* "frozen" key exists only in kernels >v5.2 so assume thawed cgroup before that. */
+        if (r != ENXIO && r < 0)
```

Comment:
ENXIO → -ENXIO

Reply:
hmm, if the call fails because an attribute is missing then ENXIO is returned and the function fails and doesn't touch `values`. This means on old kernels that have the populated field, but do not have the frozen field this will fail and not the populated state will not be iitialized either, and that's a problem...

not sure what the best approach here is... maybe just change cg_get_keyed_attribute() so that instead of returning ENXIO if any specified attribute is missing it initializes missing attributes to NULL + returns the number of initialized fields as return value. Our current users of cg_get_keyed_attributes() all pass a single attribute only so far, hence it should be straight-forward to change the call like that. Of course that means all users like that should check the return value carefully then and consider a return value of 0 like they prevously considered -ENXIO, and only > 0 means "field properly found".